game/hud/src/components/HUD/HUDEditor.tsx

```diff
@@ -0,0 +1,463 @@
+                          const deltaY = e.deltaY;
+                          setTimeout(() => this.setScale(
+                            this.props.selectedWidget.name,
+                            this.props.selectedWidget.widget,
+                            this.props.selectedWidget.widget.position.scale + (deltaY < 0 ? 0.01 : -0.01)),
+                          );
+                        }}>
+                      <i className='fa fa-search-plus'></i>
+                    </div>
+                  </a>
+                  <div className='HUDToolbarScaleControls'>
+                    <a href='#'>
+                      <div
+                        onClick={() => this.setScale(
+                          this.props.selectedWidget.name,
+                          this.props.selectedWidget.widget,
+                          this.props.selectedWidget.widget.position.scale - this.state.scaleFactor,
+                          )}
+                        onMouseDown={e => this.startScaleHold(false)}
+                      >-</div>
+                    </a>
+                    <div></div>
+                    <a href='#'>
+                      <div
+                        onClick={() => this.setScale(
+                          this.props.selectedWidget.name,
+                          this.props.selectedWidget.widget,
+                          this.props.selectedWidget.widget.position.scale + this.state.scaleFactor,
+                        )}
+                        onMouseDown={e => this.startScaleHold(true)}
+                      >+</div>
+                    </a>
+                  </div>
+                  <div className='HUDToolbarScaleText'>
+                    {`${(this.props.selectedWidget.widget.position.scale * 100).toFixed(0)}%`}
+                  </div>
+                </HUDEditorToolbarItem>
+            }
+
+            <HUDEditorToolbarItem
+                onMouseOver={ (e: React.MouseEvent<HTMLElement>) => {
+                  this.tooltipMessage = 'Reset Widget';
+                  this.tooltipEvent = e;
+                  this.onMouseOver(e);
+                } }
+                onMouseLeave={ this.onMouseLeave }
+            >
+              <a href='#'>
+                <i
+                  className='fa fa-retweet'
+                  onClick={ () => this.resetWidget(this.props.selectedWidget.name) }
+                ></i>
+              </a>
+            </HUDEditorToolbarItem>
+
+          </HUDEditorToolbar>
+        </div> }
+      </HUDEditorContainer>
+    );
+  }
+
+  public componentDidMount() {
+    window.addEventListener('mouseup', this.onMouseUp);
+    window.addEventListener('mousemove', this.onMouseMove);
+    this.props.setSelectedWidget(null);
+  }
+
+  public componentDidUnMount() {
+    window.removeEventListener('mouseup', this.onMouseUp);
+    window.removeEventListener('mousemove', this.onMouseMove);
+  }
+
+  private onMouseDown = (e: any, mode: EditMode) => {
```

Comment:
We should give the event a type, probably `React.MouseEvent<HTMLDivElement>` or something.

Reply:
✔️